addons/purchase/controllers/portal.py

```diff
@@ -196,3 +201,34 @@ def portal_my_purchase_order_update_dates(self, order_id=None, access_token=None
         if updated_dates:
             order_sudo._update_date_planned_for_lines(updated_dates)
         return Response(status=204)
+
+    @http.route(['/my/purchase/<int:order_id>/download_edi'], auth="public", website=True)
+    def portal_my_purchase_order_download_edi(self, order_id=None, access_token=None, **kw):
+        """User update scheduled date on purchase order line.
+        """
+        try:
+            order_sudo = self._document_check_access('purchase.order', order_id, access_token=access_token)
+        except (AccessError, MissingError):
+            return request.redirect('/my')
+
+        builders = []
+        order_sudo._get_edi_builders(builders)
+
+        if len(builders) != 1:
+            return request.redirect('/my')
+
+        # This handles only one builder for now, more can be added in the future
+        builder = builders[0]
+        xml_content = builder._export_order(order_sudo)
+        xml_file_data = io.BytesIO(xml_content)
+
+        download_name = builder._export_purchase_order_filename(order_sudo)
+
+        return send_file(
+            xml_file_data,
+            request.httprequest.environ,
+            download_name=download_name,
+            mimetype='text/xml',
+            as_attachment=True,
+            response_class=Response
+        )
```

Comment:
I think it's a better practice to create the attachment and return the http route to download it